crates/iceberg-rest-server/src/service/event_publisher.rs

```diff
@@ -40,10 +40,45 @@ impl EventPublisher for TracingPublisher {
         &self,
         id: Uuid,
         typ: &str,
-        data: impl Serialize + Send,
+        data: serde_json::Value,
         metadata: Vec<(String, String)>,
-    ) {
+    ) -> anyhow::Result<()> {
         let data = serde_json::to_string(&data).unwrap_or("Serialization failed".to_string());
         tracing::info!("Received event of type: '{typ}' with id: '{id}', data: '{data}' and metadata: '{metadata:?}'");
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct NatsPublisher {
+    pub client: async_nats::Client,
+    pub topic: String,
+}
+
+#[async_trait::async_trait]
+impl EventPublisher for NatsPublisher {
+    async fn publish(&self, id: Uuid, typ: &str, data: serde_json::Value, metadata: Vec<(String, String)>,
+    ) -> anyhow::Result<()> {
+        use cloudevents::{EventBuilder, EventBuilderV10};
+
+        let event_builder = EventBuilderV10::new()
```

Comment:
I think we should move this logic to the EventPublisher or introduce a separate function or layer.
We want the exact same cloud event to be sent to Kafka and REST APIs later.